File: Cosmos-Android/app/src/main/java/wannabit/io/cosmostaion/model/StarnameAccount.java
package wannabit.io.cosmostaion.model;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

public class StarNameAccount {
    @SerializedName("domain")
    public String domain;

    @SerializedName("name")
    public String name;

    @SerializedName("owner")
    public String owner;

    @SerializedName("valid_until")
    public long valid_until;

    @SerializedName("resources")
    public ArrayList<StarNameResource> resources;

    @SerializedName("certificates")
    public String certificates;

    @SerializedName("broker")
    public String broker;

    @SerializedName("metadata_uri")
    public String metadata_uri;

    public String getResourceSize() {
        if (resources == null || resources.size() == 0) {
            return "0";
        } else {
            return "" + resources.size();
        }
    }
}

File: Cosmos-IOS/Cosmostation/Cosmostation/Model/StarNameAccount.swift
//
//  StarNameAccount.swift
//  Cosmostation
//
//  Created by 정용주 on 2020/10/12.
//  Copyright © 2020 wannabit. All rights reserved.
//

import Foundation

public struct StarNameAccount: Codable {
    var domain: String = ""
    var name: String = ""
    var owner: String = ""
    var valid_until: Int64 = -1
    var certificates: String = ""
    var broker: String = ""
    var metadata_uri: String = ""
    var resources: Array<StarNameResource> = Array<StarNameResource>()
    
    init() {}
    
    init(_ dictionary: [String: Any]) {
        self.domain = dictionary["domain"] as? String ?? ""
        self.name = dictionary["name"] as? String ?? ""
        self.owner = dictionary["owner"] as? String ?? ""
        self.valid_until = dictionary["valid_until"] as? Int64 ?? -1
        self.certificates = dictionary["certificates"] as? String ?? ""
        self.broker = dictionary["broker"] as? String ?? ""
        self.metadata_uri = dictionary["metadata_uri"] as? String ?? ""
        if let rawResources = dictionary["resources"] as? Array<NSDictionary> {
            self.resources.removeAll()
            for rawResource in rawResources {
                self.resources.append(StarNameResource(rawResource as! [String : Any]))
            }
        }
    }
}
